ansible_collections/arista/avd/plugins/module_utils/eos_designs_facts.py

```diff
@@ -1185,14 +1204,24 @@ def uplinks(self):
                     uplink['peer_channel_group_id'] = ''.join(re.findall(r'\d', uplink_switch_interfaces[0]))
 
                 # Remove vlans if upstream switch does not have them #}
+                if self.enable_trunk_groups:
+                    uplink['trunk_groups'] = ["UPLINK"]
+                    if self.mlag is True:
+                        uplink['peer_trunk_groups'] = [self.group]
+                    else:
+                        uplink['peer_trunk_groups'] = [self.hostname]
+
                 switch_vlans = self._vlans
                 uplink_switch_vlans = uplink_switch_facts._vlans
                 uplink_vlans = list(set(switch_vlans).intersection(uplink_switch_vlans))
 
                 if self.inband_management_vlan is not None:
                     uplink_vlans.append(int(self.inband_management_vlan))
 
-                uplink['vlans'] = self._list_compress(uplink_vlans)
+                if uplink_vlans:
+                    uplink['vlans'] = self._list_compress(uplink_vlans)
+                else:
+                    uplink['vlans'] = 'none'
```

Comment:
I raised this concern on the chat. I don't think this will affect any deployments, since it only print out `none` in the case of _no_ vlans are configured on the child switch.
The difference on existing deployments would be less unneeded flooding. IMO this is a bugfix.

Reply:
Implemented in PR 1879. I will rebase this one